src/drivers/hermes/hermes.c

```diff
@@ -85,11 +186,45 @@ static ssize_t hermes_read(struct file *filp, char __user *buf, size_t len,
 static ssize_t hermes_write(struct file *filp, const char __user *buf,
 		size_t len, loff_t *off)
 {
-	return len;
+	int ret = 0;
+	struct hermes_cmd_req req;
+
+	if (len != sizeof(req)) {
+		ret = -ENOSPC;
```

Comment:
Not sure if this is the best error here. It kind of makes sense for len > sizeof(req), but seems odd for len < sizeof(req)

Reply:
EINVAL would be appropriate not ENOSPC... You're not writing to a full disk...